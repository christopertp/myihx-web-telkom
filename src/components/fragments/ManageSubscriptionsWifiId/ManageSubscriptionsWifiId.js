import React, { useState, useEffect, useContext } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Bullet from '../../elements/Bullet';
import Button from '../../elements/Button';
import Card from '../../elements/Card';
import Heading from '../../elements/Heading';
import IconButton from '../../elements/IconButton';
import Modal from '../../elements/Modal';
import ModalConfirmation from '../../elements/ModalConfirmation';
import FormChangeDeviceName from '../../forms/ManageSubscriptionsDeviceName';
import FormChangeChangePasswordWifiId from '../../forms/ManageSubscriptionsPasswordWifiId';
import WhiteCardPlain from '../../layouts/WhiteCardPlain';
import { AppContext } from '../../../contexts';
import styles from './styles.scoped.css';

export default function ManageSubscriptionsWifiId() {
  const { fetchData } = useSelector((s) => s.manageSubscriptions);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchData('wifiId', '122510249142'));
  }, []);
  return (
    <WhiteCardPlain back="/manage-subscriptions" title="Manage your wifi.id">
      <div className={styles['root']}>
        <LoginInformation />
        <Devices />
        <HowToWifiId />
        <Button className={styles['btn-submit']} onClick={() => {}} type="submit">
          Add A New Device
        </Button>
        <Overlay />
      </div>
    </WhiteCardPlain>
  );
}

export function LoginInformation() {
  const [showPassword, setShowPassword] = useState(false);
  const [changePassword, setChangePassword] = useState(false);
  const { wifiId, isLoading, message, fetchSubmit } = useSelector((s) => s.manageSubscriptions);
  const dispatch = useDispatch();
  const onSubmit = (value) => {
    dispatch(
      fetchSubmit(
        'updateAccountWifiId',
        {
          indihomeNumber: 122510249142,
          data: { password: value.passwordWifiId },
        },
        'wifiId',
      ),
    );
    setChangePassword(false);
  };

  const setContent = (value) => {
    if (isLoading.d) {
      return <span className="loading" />;
    } else if (!isLoading.d && message && !wifiId) {
      return '';
    }
    return value;
  };

  return (
    <section className={styles['login-information']}>
      <Heading>Your login information</Heading>
      <Card className={styles['card-login-information']} variant="hover">
        <div className={styles['field-username']}>
          <small>Username</small>
          {setContent(<p>{wifiId.username}</p>)}
        </div>
        <div className={styles['field-password']}>
          <small>Password</small>
          {setContent(
            <input
              name="password"
              readOnly
              type={showPassword ? 'text' : 'password'}
              value={wifiId.password}
            />,
          )}
          <button onClick={() => setShowPassword(!showPassword)}>
            {!showPassword ? 'Show Password' : 'Hide Password'}
          </button>
          <IconButton name="edit" onClick={() => setChangePassword(true)} />
        </div>
      </Card>
      <Modal onClose={() => setChangePassword(false)} open={changePassword || isLoading.s}>
        <FormChangeChangePasswordWifiId
          handleCancel={() => setChangePassword(false)}
          message={message}
          onSubmit={onSubmit}
        />
      </Modal>
    </section>
  );
}

export function Devices() {
  const [confirmRemoveDevice, setConfirmRemoveDevice] = useState(false);
  const [changeDeviceName, setChangeDeviceName] = useState('');
  const { wifiId, isLoading, fetchSubmit, message } = useSelector((s) => s.manageSubscriptions);
  const dispatch = useDispatch();
  const subtitle = `Are you sure you want to remove this device?
    If you want to add this device again later, you will be billed again.
    You can still login with this device until 1st of November 2019`;
  const setContent = (value) => {
    if (isLoading.d) {
      return Array(3)
        .fill(3)
        .map((val, idx) => <span className="loading" key={idx} />);
    } else if (!isLoading.d && wifiId.deviceList.length === 0) {
      return (
        <Card className={styles['card-empty-device']}>
          <img alt="chevron red" src="/assets/ic_selection_device_computer_grey.svg" />
          <span>You don't have any device</span>
          <p>Add device to use wifi.id</p>
        </Card>
      );
    }
    return value;
  };
  const onSubmit = (value) => {
    dispatch(
      fetchSubmit(
        'updateDeviceWifiId',
        {
          indihomeNumber: 122510249142,
          data: {
            crmid: changeDeviceName,
            deviceName: value.deviceName,
          },
        },
        'wifiId',
      ),
    );
    setChangeDeviceName('');
  };
  return (
    <section className={styles['devices']}>
      <Heading>Your device(s)</Heading>
      {setContent(
        wifiId.deviceList.map((value, index) => (
          <Card className={styles['card-device']} key={index}>
            <img alt="chevron red" src="/assets/ic_selection_device_computer_r.svg" />
            <span>{value.deviceName}</span>
            <p>Added {value.startDate}</p>
            <div className={styles['more']}>
              <IconButton name="more" />
              <div className={styles['more-menu']}>
                <ul>
                  <li
                    onClick={() => {
                      setChangeDeviceName(value.crmId);
                    }}>
                    Edit device name
                  </li>
                  <li
                    onClick={() => {
                      setConfirmRemoveDevice(true);
                    }}>
                    Remove device
                  </li>
                </ul>
              </div>
              <Modal
                onClose={() => setChangeDeviceName('')}
                open={Boolean(changeDeviceName || isLoading.s)}>
                <FormChangeDeviceName
                  handleCancel={() => setChangeDeviceName('')}
                  message={message}
                  onSubmit={onSubmit}
                />
              </Modal>
              <ModalConfirmation
                onAccept={() => {}}
                onClose={() => setConfirmRemoveDevice(false)}
                onRefuse={() => setConfirmRemoveDevice(false)}
                open={confirmRemoveDevice}
                subtitle={subtitle}
                title="Are you sure you want to remove this device?"
              />
            </div>
          </Card>
        )),
      )}
    </section>
  );
}

export function HowToWifiId() {
  const [lng] = useState('id');
  const { wifiId, isLoading } = useSelector((s) => s.manageSubscriptions);
  const setContent = (value) => {
    if (isLoading.d) {
      return Array(3)
        .fill(3)
        .map((val, idx) => (
          <div className={styles.loading} key={idx}>
            <span className="loading" />
            <span className="loading" />
            <span className="loading" />
            <span className="loading" />
          </div>
        ));
    } else if (!isLoading.d && wifiId.guide.length === 0) {
      return [];
    }
    return value;
  };
  return (
    <section className={styles['how-to-wifiId']}>
      {setContent(
        wifiId.guide.map((value, index) => (
          <section className={styles['steps']} key={index}>
            <Heading>{value[lng]?.header}</Heading>
            <ul>
              <li>
                <Bullet />
                <strong>{value[lng]?.description}</strong>
                {value[lng]?.steps.map((val, indx) => (
                  <p key={indx}>{val}</p>
                ))}
              </li>
            </ul>
          </section>
        )),
      )}
    </section>
  );
}

export function Overlay() {
  const { setOverlay } = useContext(AppContext);
  const { isLoading } = useSelector((s) => s.manageSubscriptions);

  useEffect(() => {
    setOverlay(isLoading.s);
  }, [isLoading.s]);

  return null;
}
