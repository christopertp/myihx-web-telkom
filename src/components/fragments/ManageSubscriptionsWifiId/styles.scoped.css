.devices {
  grid-area: devices;
}
.devices > .card-empty-device,
.devices > .card-device {
  display: grid;
  grid-template-columns: auto 1fr auto;
  margin-top: 0.55rem;
  padding: 0.85rem 0.625rem;
  position: relative;
}
.devices > span {
  border-radius: 0.5rem;
  display: block;
  height: 4.938rem;
  margin-top: 0.55rem;
}
.card-empty-device > img,
.card-device > img:first-child {
  align-self: center;
  grid-column: 1;
  grid-row: 1/3;
  height: 3.5rem;
  margin-right: 0.813rem;
}
.card-device > .more {
  display: grid;
  outline: none;
}
.card-device > .more > button > img,
.card-device > .more > button:focus {
  outline: none;
}
.card-device > .more > button {
  align-self: center;
  grid-column: 3;
  grid-row: 1/3;
}
.card-empty-device > span,
.card-device > span {
  align-self: flex-end;
  font-weight: 700;
}
.card-empty-device > p,
.card-device > p {
  color: #66686a;
  font-size: 0.813rem;
  grid-column: 2;
}
.card-device > div {
  align-self: center;
  grid-column: 3;
  grid-row: 1/3;
}
.more-menu:focus {
  outline: none;
}
.more-menu {
  background-color: #fff;
  border-radius: 0.438rem;
  box-shadow: 0 0.25rem 0.5rem rgba(0, 0, 1, 0.075);
  left: 10.3rem;
  opacity: 0;
  position: absolute;
  top: 4rem;
  transition: all ease-in-out 0.25s;
  visibility: hidden;
  width: auto;
  z-index: 2;
}
.more:hover > div {
  opacity: 1;
  visibility: visible;
}
.more-menu::before {
  background-color: white;
  bottom: 6rem;
  box-shadow: 0 0.25rem 0.5rem rgba(0, 0, 1, 0.075);
  clip: rect(0rem, 1.12rem, 1.12rem, -0.25rem);
  content: '';
  height: 1.25rem;
  position: absolute;
  right: 0.74rem;
  transform: rotate(45deg);
  width: 1.25rem;
  z-index: 0;
}
.more-menu > ul > li:nth-child(1) {
  border-bottom: 0.063rem solid #fafafa;
  padding: 0.938rem 1.25rem;
}
.more-menu > ul > li:nth-child(2) {
  color: #ee3124;
  padding: 0.938rem 1.25rem;
}
.how-to-wifiId {
  grid-area: how-to;
}
.how-to-wifiId .steps {
  margin-bottom: 1rem;
}
.how-to-wifiId .steps:last-child {
  margin-bottom: 0rem;
}
.how-to-wifiId .steps > h3 {
  margin-bottom: 0.55rem;
}
.how-to-wifiId > .steps > ul > li {
  display: grid;
  grid-template-columns: 2rem auto;
  grid-template-rows: auto auto;
  max-width: 36rem;
}
.how-to-wifiId > .steps > ul > li > div {
  grid-row: 1/3;
  margin-right: auto;
  padding-top: 0.5rem;
}
.how-to-wifiId > .steps > ul > li > h5 {
  font-size: 1.2em;
  margin: auto 0 0.3rem;
}
.how-to-wifiId > .steps > ul > li > p:first-of-type {
  margin-top: 0.3rem;
}
.how-to-wifiId > .steps > ul > li > p {
  color: #66686a;
  font-size: 0.95rem;
  grid-column: 2;
  margin-bottom: 0.625rem;
  margin-top: 0rem;
}
.how-to-wifiId > .loading {
  display: grid;
  grid-template-columns: 2rem auto;
  grid-template-rows: auto auto;
}
.how-to-wifiId > .loading > span:nth-child(1) {
  border-radius: 0.2rem;
  display: block;
  height: 1.5rem;
  width: 16rem;
}
.how-to-wifiId > .loading > span:nth-child(2) {
  border-radius: 50%;
  display: block;
  grid-row: 1/3;
  height: 1.7rem;
  width: 1.7rem;
}
.how-to-wifiId > .loading > span:nth-child(3) {
  border-radius: 0.2rem;
  display: block;
  height: 1.5rem;
  width: 18rem;
}
.how-to-wifiId > .loading > span:nth-child(4) {
  border-radius: 0.2rem;
  grid-column: 2;
  height: 1.5rem;
}

.root {
  align-items: baseline;
  display: grid;
  grid-gap: 0.625rem;
  grid-template-areas: 'login-information devices' 'how-to devices' '. devices' 'btn-submit btn-submit';
  grid-template-columns: 1fr 1fr;
  grid-template-rows: auto auto;
}
.field-password {
  display: grid;
  grid-template-columns: auto 1fr;
  grid-template-rows: auto 1fr auto;
}
.field-password > span {
  border-radius: 0.2rem;
  grid-column: 1;
  grid-row: 2;
  height: 1.2rem;
  width: 10rem;
}
.field-password > input {
  border: none;
  font-size: 1rem;
  grid-column: 1;
  grid-row: 2;
  margin-bottom: 0.5rem;
  outline: none;
}
.field-password > button {
  background-color: #fff;
  border: none;
  color: #ee3124;
  cursor: pointer;
  font-size: 0.875rem;
  font-weight: 700;
  grid-column: 1;
  grid-row: 3;
  margin-top: 0.313rem;
  text-align: start;
  width: max-content;
}
.field-password > button:last-child {
  grid-column: 2;
  grid-row: 2/4;
  justify-self: right;
}
.field-password > img,
.devices > .card-device > div:last-child {
  cursor: pointer;
}
.field-username {
  display: flex;
  flex-direction: column;
  margin-bottom: 2rem;
}
.field-username > p {
  word-break: break-word;
}

.field-username > span {
  border-radius: 0.2rem;
  height: 1.2rem;
  width: 10rem;
}
.login-information {
  grid-area: login-information;
  margin-bottom: 1rem;
  margin-top: 0.75rem;
}
.login-information > div {
  margin-top: 0.7rem;
}
.root > .btn-submit {
  grid-area: btn-submit;
}
.root > .login-information > .card-login-information small {
  color: #66686a;
}
.loading > span {
  border-radius: 0.5rem;
  display: block;
  height: 4.938rem;
  margin-top: 0.55rem;
}

@media only screen and (max-width: 48em) {
  .card-empty-device > img,
  .card-device > img:first-child {
    height: 3rem;
  }
  .card-devices > img:last-child {
    height: 1.2rem;
  }
  .card-empty-device > p,
  .card-device > p {
    align-self: center;
  }
  .devices,
  .card-empty-devices {
    margin-bottom: 1.5rem;
  }
  .more-menu {
    left: 11.5rem;
    top: 3.8rem;
  }
  .more-menu::before {
    bottom: 5rem;
    left: 6.9rem;
  }
  .more-menu > ul > li:nth-child(1),
  .more-menu > ul > li:nth-child(2) {
    font-size: 0.813rem;
  }

  .how-to-wifiId > .steps > ul > li > div {
    padding-top: 0.3rem;
  }
  .how-to-wifiId > .steps > ul > li > h5 {
    font-size: 0.8rem;
    line-height: 1.5;
  }
  .how-to-wifiId > .steps > ul > li > p {
    font-size: 0.75rem;
    line-height: 1.2rem;
  }
  .root {
    display: block;
  }
  .field-password > div:first-child > input {
    margin-bottom: 0.9rem;
  }
  .login-information > div {
    padding: 0.95rem 1rem;
  }
  .login-information > div:first-child {
    margin-top: 0;
  }
  .root > .btn-submit {
    margin-top: 1rem;
    width: 100%;
  }
  .login-information > .card-login-information small {
    font-size: 0.75rem;
  }
}
