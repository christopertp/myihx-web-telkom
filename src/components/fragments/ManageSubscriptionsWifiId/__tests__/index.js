import React, { useState, useContext } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import ManageSubscriptionsWifiId, {
  Overlay,
  LoginInformation,
  Devices,
  HowToWifiId,
} from '../ManageSubscriptionsWifiId';

useSelector.mockImplementation((fn) => {
  fn({});
  return {
    wifiId: {
      username: '',
      password: '',
      deviceList: [],
      guide: [],
    },
    isLoading: { d: false, s: false },
    fetchData: jest.fn(),
    message: '',
  };
});

describe('src/components/fragments/ManageSubscriptionsWifiId', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<ManageSubscriptionsWifiId />);
    expect(tree).toMatchSnapshot();

    const result = ManageSubscriptionsWifiId();
    const { children } = result.props;
    children.props.children[3].props.onClick();

  });

  test('render LoginInformation', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<LoginInformation />);
    expect(tree).toMatchSnapshot();

    const dispatch = jest.fn();
    const setShowPassword = jest.fn();
    const setChangePassword = jest.fn();
    useState
      .mockImplementationOnce((v) => [v, setShowPassword])
      .mockImplementationOnce((v) => [v, setChangePassword]);
    useDispatch.mockImplementation(() => dispatch);
    useSelector.mockImplementationOnce(() => ({
      message: '',
      fetchData: jest.fn(),
      fetchSubmit: jest.fn(),
      isLoading: { d: true },
      wifiId: {},
    }));

    const result = LoginInformation();
    expect(result.props.children[1].props.children[0].props.children[1].props.className).toBe(
      'loading',
    );

    result.props.children[1].props.children[1].props.children[2].props.onClick();
    expect(setShowPassword).toHaveBeenNthCalledWith(1, true);

    result.props.children[1].props.children[1].props.children[3].props.onClick();
    expect(setChangePassword).toHaveBeenNthCalledWith(1, true);

    result.props.children[2].props.onClose();
    expect(setChangePassword).toHaveBeenNthCalledWith(2, false);

    result.props.children[2].props.children.props.handleCancel();
    expect(setChangePassword).toHaveBeenNthCalledWith(3, false);

    result.props.children[2].props.children.props.onSubmit('testpassword');
    expect(setChangePassword).toHaveBeenNthCalledWith(4, false);
  });

  test('render LoginInformation with error', () => {
    const dispatch = jest.fn();
    const setShowPassword = jest.fn();
    const setChangePassword = jest.fn();
    useState
      .mockImplementationOnce(() => [true, setShowPassword])
      .mockImplementationOnce(() => [true, setChangePassword]);
    useDispatch.mockImplementation(() => dispatch);
    useSelector.mockImplementationOnce(() => ({
      message: 'test error',
      fetchData: jest.fn(),
      fetchSubmit: jest.fn(),
      isLoading: { d: false },
      wifiId: false,
    }));

    const result = LoginInformation();
    expect(result.props.children[1].props.children[0].props.children[1]).toBe('');
  });

  test('render Devices', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Devices />);
    expect(tree).toMatchSnapshot();

    const dispatch = jest.fn();
    const setChangeDeviceName = jest.fn();
    const setConfirmRemoveDevice = jest.fn();
    useState
      .mockImplementationOnce((v) => [v, setConfirmRemoveDevice])
      .mockImplementationOnce((v) => [v, setChangeDeviceName]);
    useDispatch.mockImplementation(() => dispatch);
    useSelector.mockImplementationOnce(() => ({
      message: '',
      fetchData: jest.fn(),
      fetchSubmit: jest.fn(),
      isLoading: { d: true },
      wifiId: {
        deviceList: [],
      },
    }));

    const result = Devices();

    expect(result.props.children[1].length).toBe(3);
    expect(result.props.children[1][0].props.className).toBe('loading');
  });

  test('render Devices with data', () => {
    const dispatch = jest.fn();
    const setChangeDeviceName = jest.fn();
    const setConfirmRemoveDevice = jest.fn();
    useState
      .mockImplementationOnce((v) => [v, setConfirmRemoveDevice])
      .mockImplementationOnce((v) => [v, setChangeDeviceName]);
    useDispatch.mockImplementation(() => dispatch);
    useSelector.mockImplementationOnce(() => ({
      message: '',
      fetchData: jest.fn(),
      fetchSubmit: jest.fn(),
      isLoading: { d: false },
      wifiId: {
        deviceList: [
          {
            deviceName: 'tesAja8',
            crmId: 'SEAMLESS1',
            startDate: '18 Jun 2020',
          },
        ],
      },
    }));

    const result = Devices();
    const { children } = result.props;
    expect(children[1][0].props.children[1].props.children).toBe('tesAja8');
    children[1][0].props.children[3].props.children[1]
      .props.children.props.children[0].props.onClick();
    expect(setChangeDeviceName).toHaveBeenNthCalledWith(1, 'SEAMLESS1');

    children[1][0].props.children[3].props.children[1]
      .props.children.props.children[1].props.onClick();
    expect(setConfirmRemoveDevice).toHaveBeenNthCalledWith(1, true);

    result.props.children[1][0].props.children[3].props.children[2].props.onClose();
    expect(setChangeDeviceName).toHaveBeenNthCalledWith(2,'');

    result.props.children[1][0].props.children[3].props.children[2]
      .props.children.props.handleCancel();
    expect(setChangeDeviceName).toHaveBeenNthCalledWith(2,'');

    result.props.children[1][0].props.children[3].props.children[2]
      .props.children.props.onSubmit('tesAja8');
    expect(setChangeDeviceName).toHaveBeenNthCalledWith(2,'');

    result.props.children[1][0].props.children[3].props.children[3].props.onClose();
    expect(setConfirmRemoveDevice).toHaveBeenNthCalledWith(2,false);

    result.props.children[1][0].props.children[3].props.children[3].props.onRefuse();
    expect(setConfirmRemoveDevice).toHaveBeenNthCalledWith(2,false);

    result.props.children[1][0].props.children[3].props.children[3].props.onAccept();
  });

  test('render HowToWifiId', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<HowToWifiId />);

    expect(tree).toMatchSnapshot();
    const dispatch = jest.fn();
    useDispatch.mockImplementation(() => dispatch);
    useSelector.mockImplementationOnce(() => ({
      message: '',
      fetchData: jest.fn(),
      fetchSubmit: jest.fn(),
      isLoading: { d: true },
      wifiId: {
        guide: [],
      },
    }));

    const result = HowToWifiId();
    expect(result.props.children[1].props.children[0].props.className).toBe('loading');
  });

  test('render HowToWifiId with data', () => {
    const dispatch = jest.fn();
    useDispatch.mockImplementation(() => dispatch);
    useSelector.mockImplementationOnce(() => ({
      message: '',
      fetchData: jest.fn(),
      fetchSubmit: jest.fn(),
      isLoading: { d: false },
      wifiId: {
        guide: [
          {
            'detailId': '46d05732-5850-4abe-b526-45a6e50a5b81',
            'name': 'howToUse',
            'id': {
              'header': 'test header',
              'description': 'test desc',
              'steps': [
                'test steps'
              ]
            },
            'en': {
              'header': 'test header',
              'description': 'test desc',
              'steps': []
            }
          },
        ],
      },
    }));

    const result = HowToWifiId();
    expect(result.props.children[0].props.children[0].props.children).toBe('test header');
    expect(result.props.children[0].props.children[1]
      .props.children.props.children[0].key).toBe(null);
    expect(result.props.children[0].props.children[1]
      .props.children.props.children[2][0].props.children).toBe('test steps');
  });

  test('Overlay', () => {
    const setOverlay = jest.fn();
    useContext.mockImplementationOnce(() => ({ setOverlay }));
    Overlay();
    expect(setOverlay).toHaveBeenCalledWith(false);
  });
});
