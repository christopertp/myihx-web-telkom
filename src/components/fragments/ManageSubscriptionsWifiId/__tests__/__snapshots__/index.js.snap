// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/ManageSubscriptionsWifiId render 1`] = `
<WhiteCardPlain
  back="/manage-subscriptions"
  close=""
  help={false}
  icon=""
  size="large"
  subtitle=""
  title="Manage your wifi.id"
>
  <div>
    <LoginInformation />
    <Devices />
    <HowToWifiId />
    <Button
      className=""
      disabled={false}
      onClick={[Function]}
      to=""
      type="submit"
      variant="red"
    >
      Add A New Device
    </Button>
    <Overlay />
  </div>
</WhiteCardPlain>
`;

exports[`src/components/fragments/ManageSubscriptionsWifiId render Devices 1`] = `
<section>
  <Heading
    className=""
  >
    Your device(s)
  </Heading>
  <Card
    className=""
    disabled={false}
    onClick={null}
    to=""
    variant="bordered"
  >
    <img
      alt="chevron red"
      src="/assets/ic_selection_device_computer_grey.svg"
    />
    <span>
      You don't have any device
    </span>
    <p>
      Add device to use wifi.id
    </p>
  </Card>
</section>
`;

exports[`src/components/fragments/ManageSubscriptionsWifiId render HowToWifiId 1`] = `<section />`;

exports[`src/components/fragments/ManageSubscriptionsWifiId render LoginInformation 1`] = `
<section>
  <Heading
    className=""
  >
    Your login information
  </Heading>
  <Card
    className=""
    disabled={false}
    onClick={null}
    to=""
    variant="hover"
  >
    <div>
      <small>
        Username
      </small>
      <p>
        
      </p>
    </div>
    <div>
      <small>
        Password
      </small>
      <input
        name="password"
        readOnly={true}
        type="password"
        value=""
      />
      <button
        onClick={[Function]}
      >
        Show Password
      </button>
      <IconButton
        className=""
        name="edit"
        onClick={[Function]}
        title=""
        to=""
        type="button"
      />
    </div>
  </Card>
  <Modal
    className=""
    onClose={[Function]}
    open={false}
  >
    <ReduxForm
      destroyOnUnmount={true}
      enableReinitialize={false}
      forceUnregisterOnUnmount={false}
      form="manageSubscriptionsPasswordWifiId"
      getFormState={[Function]}
      handleCancel={[Function]}
      initialValues={
        Object {
          "passwordWifiId": "",
        }
      }
      keepDirtyOnReinitialize={false}
      message=""
      onSubmit={[Function]}
      persistentSubmitErrors={false}
      pure={true}
      shouldAsyncValidate={[Function]}
      shouldError={[Function]}
      shouldValidate={[Function]}
      shouldWarn={[Function]}
      submitAsSideEffect={false}
      touchOnBlur={true}
      touchOnChange={false}
      updateUnregisteredFields={false}
      validate={[Function]}
    />
  </Modal>
</section>
`;
